vernac/vernacentries.mli

```diff
@@ -20,10 +20,14 @@ val vernac_require :
   Libnames.qualid option -> bool option -> Libnames.qualid list -> unit
 
 (** The main interpretation function of vernacular expressions *)
-val interp :
-  ?verbosely:bool ->
-  ?proof:(Proof_global.proof_object * Lemmas.Info.t) ->
-  st:Vernacstate.t -> Vernacexpr.vernac_control -> Vernacstate.t
+val interp : ?verbosely:bool -> st:Vernacstate.t -> Vernacexpr.vernac_control -> Vernacstate.t
+
+val interp_qed_inject_proof
```

Comment:
Maybe this should live in stm.ml itself, then we only have to expose the 2 functions in lemmas.ml for this special stuff.

Reply:
I thought about that but indeed the API can be used by other clients not only the STM, it is the only way we can delay an opaque proof from above so it makes sense to expose it.

Also some other stuff such as cache handling could maybe have to go there. I dunno.